modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy

```diff
@@ -647,42 +647,104 @@ class PodSpecBuilder {
         return res
     }
 
+
     @PackageScope
-    String getAcceleratorType(AcceleratorResource accelerator) {
+    void validateAccelerator(AcceleratorResource accelerator) {
+        // gpu-small: nvidia-t4 (1)
+        // gpu-large: nvidia-a10g (1)
+        // v100-x1: nvidia-v100 (1)
+        // v100-x4: nvidia-v100 (4)
+        // v100-x8: nvidia-v100 (8)
+
+        if (accelerator.type == null) {
+            accelerator.setProperty("type", "nvidia-t4")
+            log.info("No GPU type specified - defaulting to \"nvidia-t4\"")
+        }
```

Comment:
i wonder if it makes sense to do this outside the validate function? makes more sense to me to have this not be part of the validation logic esp since its modifying the input